### PR TITLE
Add longphase/haplotag

### PR DESCRIPTION
**Pull request by fellen31**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/longphase/haplotag/main.nf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    input:
+    tuple val(meta), path(bam), path(bai), path(snps), path(svs), path(mods)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(fai)
+
+
+    output:
+    tuple val(meta), path("*.bam") , emit: bam , optional: true
+    tuple val(meta), path("*.cram"), emit: cram, optional: true
```
Comment (nvnieuwk) [inline modules/nf-core/longphase/haplotag/main.nf:18]: These should be the same output channel as of the new guidelines (currently still a PR, but will be in there soon: https://github.com/nf-core/website/pull/2661)

File: modules/nf-core/longphase/haplotag/main.nf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        longphase: \$(longphase --version | head -n 1 | sed 's/Version: //')
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def log = args.contains('--log') ? "touch ${prefix}.log" : ''
+    """
+    touch ${prefix}.bam
```
Comment (nvnieuwk) [inline modules/nf-core/longphase/haplotag/main.nf:58]: Can you somehow add some logic here that will create CRAM files when the main code would? (only if it's really possible)

File: modules/nf-core/longphase/haplotag/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.versions,
+                    bam(process.out.bam.get(0).get(1), stringency: 'silent').getHeader(),
+                    bam(process.out.bam.get(0).get(1), stringency: 'silent').getReadsMD5(),
```
Comment (nvnieuwk) [inline modules/nf-core/longphase/haplotag/tests/main.nf.test:43]: Awesome to see some `nft-bam` implementations :partying_face:

**Review by nvnieuwk — COMMENTED** [review]
Some comments

File: modules/nf-core/longphase/haplotag/tests/main.nf.test (hunk 3 above)
Comment (SPPearce) [reply modules/nf-core/longphase/haplotag/tests/main.nf.test:43]: What is the `stringency` actually doing here?

Comment (nvnieuwk) [reply modules/nf-core/longphase/haplotag/tests/main.nf.test:43]: I guess the BAM validation fails for those files

Comment (fellen31) [reply modules/nf-core/longphase/haplotag/tests/main.nf.test:43]: Yes, I put some test files with readgroups not in header.. wasn't aware of it before using nft-bam :grimacing:

File: modules/nf-core/longphase/haplotag/main.nf (hunk 1 above)
Comment (fellen31) [reply modules/nf-core/longphase/haplotag/main.nf:18]: Nice! What should we call the bam/cram output channel, bam?

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**